### PR TITLE
*: add a new directory BaseFile

### PR DESCRIPTION
**Pull request by Lloyd-Pottiger**
### What problem does this PR solve?

Issue Number: ref #8691

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/BaseFile/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Copyright 2023 PingCAP, Inc.
```
Comment (JaySon-Huang) [inline dbms/src/BaseFile/CMakeLists.txt:1]: I think removing this file does not affects anything

**Review by JaySon-Huang — APPROVED** [review]
LGTM with a nit comment

File: dbms/src/BaseFile/CMakeLists.txt (hunk 1 above)
Comment (Lloyd-Pottiger) [reply dbms/src/BaseFile/CMakeLists.txt:1]: `add_subdirectory` requires the subdirectory to contain a `CMakeLists.txt` file.

```
add_subdirectory (BaseFile)
```

Comment (JaySon-Huang) [reply dbms/src/BaseFile/CMakeLists.txt:1]: But actually we don't need to add the `add_subdirectory` statement

Comment (Lloyd-Pottiger) [reply dbms/src/BaseFile/CMakeLists.txt:1]: yeah, but if we need to split into smaller compile units, we need to do this, I think we just keep this.

**Pull request merged**